usage_test.go

```diff
@@ -0,0 +1,356 @@
+..ENV_CONFIG_MULTI_WITH_DIFFERENT_ALT..............String.....................................................
+..ENV_CONFIG_EMBEDDED_WITH_ALT.....................String.....................................................
+..ENV_CONFIG_DEBUG.................................Boolean....................................................
+..ENV_CONFIG_PORT..................................Integer....................................................
+..ENV_CONFIG_RATE..................................Float......................................................
+..ENV_CONFIG_USER..................................String.....................................................
+..ENV_CONFIG_TTL...................................Unsigned.Integer(32.bits}..................................
+..ENV_CONFIG_TIMEOUT...............................Duration...................................................
+..ENV_CONFIG_ADMINUSERS............................List.of.String.............................................
+..ENV_CONFIG_MAGICNUMBERS..........................List.of.Integer............................................
+..ENV_CONFIG_MULTIWORDVAR..........................String.....................................................
+..ENV_CONFIG_SOMEPOINTER...........................Reference.to.String........................................
+..ENV_CONFIG_SOMEPOINTERWITHDEFAULT................Reference.to.String..........foo2baz.......................foorbar.is.the.word
+..ENV_CONFIG_MULTI_WORD_VAR_WITH_ALT...............String.....................................................what.alt
+..ENV_CONFIG_MULTI_WORD_VAR_WITH_LOWER_CASE_ALT....String.....................................................
+..ENV_CONFIG_SERVICE_HOST..........................String.....................................................
+..ENV_CONFIG_DEFAULTVAR............................String.......................foobar........................
+..ENV_CONFIG_REQUIREDVAR...........................String.........................................true........
+..ENV_CONFIG_BROKER................................String.......................127.0.0.1.....................
+..ENV_CONFIG_REQUIREDDEFAULT.......................String.......................foo2bar...........true........
+..ENV_CONFIG_OUTER.................................Nested.Struct..............................................
+..ENV_CONFIG_OUTER_INNER...........................String.....................................................
+..ENV_CONFIG_OUTER_PROPERTYWITHDEFAULT.............String.......................fuzzybydefault................
+..ENV_CONFIG_AFTERNESTED...........................String.....................................................
+..ENV_CONFIG_HONOR.................................HonorDecodeInStruct........................................
+..ENV_CONFIG_DATETIME..............................Time.......................................................
+`
+
+var TestUsageListResult string = `..ENV_CONFIG_EMBEDDED
```

Comment:
not really a good way to do this as the format of the strings is pretty important to the tests. the sparse data idea was just going to cause the definition of a lot of structure data, which likely would not have made the files more approachable nor would it adequately capture the format issued.i moved the expected results to separate data files.